lib/auth/app.ex

```diff
@@ -0,0 +1,142 @@
+defmodule Auth.App do
+  @moduledoc """
+  Schema and helper functions for creating/managing Apps.
+  """
+  use Ecto.Schema
+  import Ecto.Changeset
+  import Ecto.Query, warn: false
+  alias Auth.Repo
+  # https://stackoverflow.com/a/47501059/1148249
+  alias __MODULE__
+
+  schema "apps" do
+    field :desc, :binary
+    field :end, :naive_datetime
+    field :name, :binary
+    field :url, :binary
+    field :person_id, :id
+    field :status, :id
+    has_many :apikeys, Auth.Apikey
+
+    timestamps()
+  end
+
+  @doc false
+  def changeset(app, attrs) do
+    app
+    |> cast(attrs, [:name, :desc, :url, :end, :person_id, :status])
+    |> validate_required([:name, :url])
+  end
+
+  @doc """
+  Returns the list of apps.
+
+  ## Examples
+
+      iex> list_apps()
+      [%App{}, ...]
+
+  """
+  def list_apps do
+    App
+    |> where([a], a.status != 6)
+    |> Repo.all()
+  end
+
+  @doc """
+  Gets a single app.
+
+  Raises `Ecto.NoResultsError` if the App does not exist.
+
+  ## Examples
+
+      iex> get_app!(123)
+      %App{}
+
+      iex> get_app!(456)
+      ** (Ecto.NoResultsError)
+
+  """
+  def get_app!(id) do
+    App
+    |> where([a], a.id == ^id and a.status != 6)
```

Comment:
Something like: 
```suggestion
    |> where([a], a.id == ^id and a.status != 6,
         preload: [:apikeys]
    )
```

Could possibly lower the amount of passes on the database

Reply:
Sadly this query didn't work. see: https://travis-ci.org/github/dwyl/auth/builds/725858363#L391
very happy to simplify _all_ the SQL/Ecto queries as a future enhancement. 👍